apollo-router/src/plugins/demand_control/mod.rs

```diff
@@ -357,11 +368,22 @@ impl Plugin for DemandControl {
                     })
                 })
                 .map_future_with_request_data(
-                    |req: &subgraph::Request| {
-                        //TODO convert this to expect
-                        req.executable_document.clone().unwrap_or_else(|| {
-                            Arc::new(Valid::assume_valid(ExecutableDocument::new()))
-                        })
+                    move |req: &subgraph::Request| {
+                        req.subgraph_name
+                            .as_ref()
+                            .and_then(|subgraph| subgraph_schemas.get(subgraph))
+                            .zip(req.operation.as_ref())
+                            .ok_or(DemandControlError::QueryParseFailure(
+                                "Cannot parse subgraph request due to missing subgraph schema"
+                                    .to_string(),
+                            ))
+                            .and_then(|(schema, operation)| {
+                                operation
+                                    .as_parsed(schema)
+                                    .map_err(DemandControlError::from)
+                            })
+                            .expect("must have a valid query")
```

Comment:
I think we should convert to an error here rather than using expect. `DemandControlError::QueryParseFailure` could be reused, or we could add something else.